packages/core/src/Popout/Popout.stories.tsx

```diff
@@ -0,0 +1,147 @@
+import { Information as InformationIcon, Pin as PinIcon, Search as SearchIcon } from 'pcln-icons'
+import React, { ChangeEvent, useEffect, useRef, useState } from 'react'
+import styled from 'styled-components'
+import { Box, Flex, FormField, getPaletteColor, Input, Label, Popout, Text } from '..'
+
+export default {
+  component: Popout,
+}
+
+const states = [
+  'Alabama',
+  'Alaska',
+  'Arizona',
+  'Arkansas',
+  'California',
+  'Colorado',
+  'Connecticut',
+  'Delaware',
+  'Florida',
+  'Georgia',
+  'Hawaii',
+  'Idaho',
+  'Illinois',
+  'Indiana',
+  'Iowa',
+  'Kansas',
+  'Kentucky',
+  'Louisiana',
+  'Maine',
+  'Maryland',
+  'Massachusetts',
+  'Michigan',
+  'Minnesota',
+  'Mississippi',
+  'Missouri',
+  'Montana',
+  'Nebraska',
+  'Nevada',
+  'New Hampshire',
+  'New Jersey',
+  'New Mexico',
+  'New York',
+  'North Carolina',
+  'North Dakota',
+  'Ohio',
+  'Oklahoma',
+  'Oregon',
+  'Pennsylvania',
+  'Rhode Island',
+  'South Carolina',
+  'South Dakota',
+  'Tennessee',
+  'Texas',
+  'Utah',
+  'Vermont',
+  'Virginia',
+  'Washington',
+  'West Virginia',
+  'Wisconsin',
+  'Wyoming',
+]
+
+const PclnSearchItem = styled(Flex)`
+  cursor: pointer;
+  white-space: nowrap;
+  &:hover,
+  &:focus {
+    background-color: ${getPaletteColor('primary.base')};
+    color: white;
+  }
+`
+
+const Wrapper = ({ children }) => <Box m={5}>{children}</Box>
+
+export const Basic = () => {
+  const ref = useRef(null)
+  const [value, setValue] = useState('')
+  const trigger = <input aria-label='x' ref={ref} value={value} onChange={(e) => setValue(e.target.value)} />
+  const content = value && <h1>{value}</h1>
+
+  return (
+    <Wrapper>
+      <Popout trigger={trigger} triggerRef={ref} content={content} />
+    </Wrapper>
+  )
+}
+
+export const IconPopout = () => {
```

Comment:
Would it make sense for clicking the trigger again when the popout is open to close the popout?

https://user-images.githubusercontent.com/3260642/205312872-1b4e73f6-606e-4a81-a212-131c8a0a843f.mp4

Reply:
Great idea, added a prop called `closeOnTriggerRefClick` for this behavior! Makes it optional, so it can be applied for cases like the Icon Example, but left off for input fields which you'd want to be able to click without closing the Popout. Thanks for the feedback!